src/stores/CheckoutController.ts

```diff
@@ -1,152 +1,174 @@
 
-  useCheckoutStore.getState().setOrder(response.responseData?.order);
+      return response.responseData?.paymentMethods;
+    });
+  });
 };
 
-export const getPaymentMethods = async () => {
-  const { cleengSandbox, cleengId } = useConfigStore.getState().getCleengData();
+export const paymentWithoutDetails = async (): Promise<unknown> => {
+  return await useAccount(async ({ auth: { jwt } }) => {
+    return await useService(async ({ checkoutService, sandbox = true, authProviderId }) => {
+      if (!checkoutService) throw new Error('checkout service is not available');
 
-  const { user, auth } = useAccountStore.getState();
-  const { paymentMethods } = useCheckoutStore.getState();
+      const { order } = useCheckoutStore.getState();
 
-  if (paymentMethods) return paymentMethods; // already fetched payment methods
-  if (!cleengId) throw new Error('cleengId is not configured');
-  if (!user || !auth) throw new Error('user is not logged in');
+      if (!order) throw new Error('No order created');
+      if (!authProviderId) throw new Error('auth provider is not configured');
 
-  const response = await checkoutService.getPaymentMethods(cleengSandbox, auth.jwt);
+      const response = await checkoutService.paymentWithoutDetails({ orderId: order.id }, sandbox, jwt);
 
-  if (response.errors.length > 0) throw new Error(response.errors[0]);
+      if (response.errors.length > 0) throw new Error(response.errors[0]);
+      if (response.responseData.rejectedReason) throw new Error(response.responseData.rejectedReason);
 
-  useCheckoutStore.getState().setPaymentMethods(response.responseData?.paymentMethods);
-
-  return response.responseData?.paymentMethods;
+      return response.responseData;
+    });
+  });
 };
 
-export const paymentWithoutDetails = async () => {
-  const { cleengSandbox, cleengId } = useConfigStore.getState().getCleengData();
-
-  const { user, auth } = useAccountStore.getState();
-  const { order } = useCheckoutStore.getState();
+export const directPostCardPayment = async (cardPaymentPayload: CardPaymentData): Promise<unknown> => {
+  return await useAccount(async ({ customer }) => {
+    return await useService(async ({ checkoutService, authProviderId }) => {
+      const { order } = useCheckoutStore.getState();
 
-  if (!order) throw new Error('No order created');
-  if (!cleengId) throw new Error('cleengId is not configured');
-  if (!user || !auth) throw new Error('user is not logged in');
+      if (!order) throw new Error('No order created');
+      if (!authProviderId) throw new Error('auth provider is not configured');
+      if (!checkoutService) throw new Error('checkout service is not available');
 
-  const response = await checkoutService.paymentWithoutDetails({ orderId: order.id }, cleengSandbox, auth.jwt);
+      // subscribe to listen to inplayer websocket notifications
+      await subscribeToNotifications(customer?.uuid);
```

Comment:
Was moving this here an effort to avoid unexpected notifications?  Imho, I think it's going to be more maintainable to subscribe upon logout still, but then check state data before action on certain messages.

Reply:
yes, we only subscribe to listen to notifications when payment is submitted. On logout we unsubscribe.

Reply:
With the useCheckAccess hook, the notifications can be excluded though, only this way the entire process is faster. We are only using the notifications for access granted, but that is handled with useCheckAccess as well. 
We should discuss if we want to have the notifications at all. But maybe at some later stage they will be needed for something.. not sure